lib/elixir/pages/getting-started/writing-documentation.md

```diff
@@ -58,80 +73,156 @@ def size(%{size: size}) do
 ```
 
-Note that the `:deprecated` key does not warn when a developer invokes the functions. If you want the code to also emit a warning, you can use the `@deprecated` attribute:
+In this example, `@spec` is the module attribute indicating that a type specification for a function will follow. `@spec` should always immediately precede the function name.``
+
+`[integer]` is the function's parameter type list. The square brackets denote a list and the integer inside the brackets means this function expects a list of integers.
+
+ The `::` symbol separates the parameter type list from the return type. It can be read as "yields" or "returns."
+
+The function `sum/1` has a return type of `integer` so we know to expect a single integer when working with this function.
+
+You can read more about types and specifications at our [Typespecs Reference](https://hexdocs.pm/elixir/1.16.0/typespecs.html#defining-a-specification).
+
+## Metadata
+
+Elixir allows developers to attach arbitrary metadata to module attributes. Metadata is added to documentation by passing a keyword list to the relevant attribute, like `@doc`, `@moduledoc`, or `@typedoc`.
+
+### `:since` Version MetaData
+
+A commonly used metadata key is `:since`. It annotates in which version that particular module, function, or type was added. 
+ 
+In the module example, the developer passes the keyword list `[:since, "1.3.0"]` to `@doc` (and `[since: "1.1.0"]` to `@moduledoc`). The `@doc` attaches the version information along with the text string, argument name and other captured data to the documentation.
+
+### `:deprecated` Deprecation MetaData
+
+Another common metadata is `:deprecated`. It emits a warning in the documentation, discouraging the use of its associated entity.
 
 ```elixir
-@deprecated "Use Foo.bar/2 instead"
+@doc deprecated: "Use MyApp.Hello.world/1 instead"
+  def hello_world(name) do
+    IO.puts("hello #{name}")
+  end
 ```
 
-Metadata can have any key. Documentation tools often use metadata to provide more data to readers and to enrich the user experience.
+Note that the `:deprecated` key only affects documentation and does not warn when a developer invokes the functions. If you want the code to also emit a warning, you can use the `@deprecated` attribute.
 
-## Recommendations
+```elixir
+@deprecated "Use MyApp.Hello.world/1 instead"
+  def hello_world(name) do
+    IO.puts("hello #{name}")
+  end
+```
 
-When writing documentation:
+## Recommendations
 
-  * Keep the first paragraph of the documentation concise and simple, typically one-line. Tools like [ExDoc](https://github.com/elixir-lang/ex_doc/) use the first line to generate a summary.
+When writing documentation, we recommend you follow these guidelines:
 
-  * Reference modules by their full name.
+  * Keep the first paragraph of the documentation a concise description, ideally one line. Tools like [ExDoc](https://github.com/elixir-lang/ex_doc/) use the first line to generate a summary.
```

Comment:
This is pretty much the same as before, but written differently. Please avoid such changes.